docs/mindsdb_sql/agents/knowledge-bases.mdx

```diff
@@ -3,238 +3,436 @@ title: Knowledge Base
+         "openai_api_version": "2024-12-01-preview"
+     },
+     reranking_model = {
+         "provider": "OpenAI",
+         "model_name": 'gpt-4o',
+         "api_key": "sk-..."  -- optional, default from env variable
+     },
+     storage = my_vector_store, -- optional, default ChromaDB
+     metadata_columns = ['date', 'creator', ...], -- optional
+     content_columns = ['review', 'content', ...], -- optional, default content
+     id_column = 'id'; -- optional
+```
 
-- `model`: This is an embedding model created within MindsDB with `CREATE MODEL embedding_model`. If you do not provide the `model` parameter, then the OpenAI's text-embedding-ada-002 model is used by default, provided that the `OPENAI_API_KEY` environment variable is defined.
-- `storage`: This is a vector database that stores embedded data. It defaults to ChromaDB, or you can connect your vector store to MindsDB with `CREATE DATABASE vector_database`.
-- `metadata_columns`: The list of column names that will be stored in the `metadata` column of the knowledge base. If not set, the `metadata` column is not used.
-- `content_columns`: The list of column names that will be stored in the `content` column of the knowledge base. If not set, all columns are stored in the `content` column.
-- `id_column`: The column name that will be stored in the `id` column of the knowledge base to uniquely identitify the data.
+</CodeGroup>
 
-<Note>
-Each knowledge base comprises the `id`, `content`, and `metadata` columns that store data defined in the parameters.
-</Note>
+Upon execution, it registers `my_kb` and associates the specified models and storage.
+
+The following is an explanation of the syntax and parameters:
+
+* `embedding_model` stores specification of the embedding model to be used.
+      * `provider` defines the model provider. Currently, the supported providers include OpenAI and its models via Azure.
+      * `model_name` defines the embedding model name as specified by the provider. Users can choose one of the [OpenAI embedding models](https://platform.openai.com/docs/guides/embeddings/embedding-models#embedding-models).
+      * `api_key` stores your OpenAI API key. Alternatively, you can use the `OPENAI_API_KEY` environment variable to store your OpenAI API key.
+      * `azure_endpoint` is required when using `Azure_OpenAI` as the provider.
+      * `openai_api_version` is required when using `Azure_OpenAI` as the provider.
 
-## Examples
+* `reranking_model` stores specification of the reranking model to be used.
+      * `provider` defines the model provider. Currently, the supported providers include OpenAI.
+      * `model_name` defines the reranking model name as specified by the provider. Users can choose one of the [OpenAI chat models](https://platform.openai.com/docs/models).
+      * `api_key` stores your OpenAI API key. Alternatively, you can use the `OPENAI_API_KEY` environment variable to store your OpenAI API key.
```

Comment:
@martyna-mindsdb In addition to the models defined here (within `embedding_model` and `reranking_model`), other compatible parameters with each API can also be passed, such as temperature etc. Do you think we need to mention this fact here?
CC: @ea-rus

Reply:
@martyna-mindsdb Actually, I am not 100% sure if these need to be mentioned or not. @ea-rus What do you think?